metrics/access-logs/sniped/main.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"k8s.io/klog"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/crane"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/google"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+// Based on gcrane list command  https://github.com/google/go-containerregistry/blob/master/cmd/gcrane/cmd/list.go
+
+func main() {
+	ctx := context.Background()
+
+	if err := run(ctx); err != nil {
+		klog.Fatalf("unexpected error: %v", err)
+	}
+}
+
+var remoteOptions []remote.Option
+
+func run(ctx context.Context) error {
+	root := "k8s.gcr.io"
+	recursive := true
+
+	remoteOptions = []remote.Option{
+		remote.WithAuthFromKeychain(authn.DefaultKeychain),
+	}
+
+	if root == "k8s.gcr.io" {
+		response, err := http.Get("https://k8s.gcr.io/v2/tags/list")
+		if err != nil {
+			return fmt.Errorf("error querying gcr root: %w", err)
+		}
+
+		b, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return fmt.Errorf("error reading gcr root response: %w", err)
+		}
+
+		root := &Root{}
+		if err := json.Unmarshal(b, root); err != nil {
+			return fmt.Errorf("error parsing gcr root: %w", err)
+		}
+
+		for _, child := range root.Child {
+			if err := dumpRepo(ctx, "k8s.gcr.io/"+child, recursive); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	if err := dumpRepo(ctx, root, recursive); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func dumpRepo(ctx context.Context, root string, recursive bool) error {
+	klog.Infof("dumping %q", root)
+
+	repo, err := name.NewRepository(root)
+	if err != nil {
+		return err
+	}
+
+	if recursive {
+		if err := google.Walk(repo, dump, google.WithAuthFromKeychain(google.Keychain)); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	tags, err := google.List(repo, google.WithAuthFromKeychain(google.Keychain))
+	if err != nil {
+		return err
+	}
+
+	return dump(repo, tags, err)
+}
+
+func dump(repo name.Repository, tags *google.Tags, err error) error {
+	if err != nil {
+		return err
+	}
+
+	var out []*Row
+
+	for _, manifest := range tags.Manifests {
+
+		for _, tag := range manifest.Tags {
+			/*
+				ref, err := name.ParseReference(digest, repo.Registry)
+				if err != nil {
+					return fmt.Errorf("parsing reference %q: %v", r, err)
+				}
+				manifest, err := remote.Get(ref, remoteOptions...)
+			*/
```

Comment:
Not a fan of checking in commented-out code.  Why was this commented out?  Dropping a TODO would be more helpful
```suggestion
```